src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenNullCoalescingAssignmentTests.cs

```diff
@@ -1418,33 +1418,26 @@ public static void Main()
         [Fact]
         public void TypeParameterLHS()
         {
-            CreateCompilation(@"
-class C
-{
-    void M<T>(T t)
-    {
-        t ??= default;
-    }
-}
-").VerifyDiagnostics(
-                // (6,9): error CS0019: Operator '??=' cannot be applied to operands of type 'T' and 'default'
-                //         t ??= default;
-                Diagnostic(ErrorCode.ERR_BadBinaryOps, "t ??= default").WithArguments("??=", "T", "default").WithLocation(6, 9)
-            );
-
             CompileAndVerify(@"
 using System;
 class C
 {
     static void Main()
     {
         Verify<object>(null, ""Assignment Evaluated"");
+        Verify<int>(default, 10);
     }
-    static void Verify<T>(T t1, T t2) where T : class
+    static void Verify<T>(T t1, T t2)
     {
         Console.WriteLine(t1 ??= t2);
+        Console.WriteLine(t1);
     }
-}", expectedOutput: "Assignment Evaluated");
+}", expectedOutput: @"
+Assignment Evaluated
+Assignment Evaluated
+0
+0
+");
```

Comment:
Consider testing `where T : class` in addition to unconstrained.

Reply:
Also verified the behavior of `where T : struct`

---
In reply to: [213789722](https://github.com/dotnet/roslyn/pull/29577#discussion_r213789722) [](ancestors = 213789722)